### DISCUSSION TITLE
Spotify and Pinterest

### DISCUSSION THREAD

**Discussion by RevealedSoulEven** [JunkFood02/Seal#414] [Ideas]
What about adding support for Spotify and Pinterest?

**Comment by Marco-9456** [comment]
> What about adding support for Pinterest?

- Pinterest is already supported by yt-dlp but only videos since yt-dlp main focus is downloading video

> What about adding support for Spotify ?

- Yt-dlp will never bring support for Spotify because it uses DRM protection, But there's a workaround that BobbyESP is working on **(take a look at his latest PR)**

**Comment by RevealedSoulEven** [comment]
I saw his project, but he just added spotifytoyoutube api where the same
songs are downloaded from youtube.

But not an issue, I use another app named 'spotoflyer' for that purpose

**Reply by BobbyESP** [reply]
Well, Spotify downloading it's possible, but this app is just a yt-dlp GUI, so for Spotify music downloading another app should be created. The implementation that I've made is to add an API that converts Spotify links into Youtube Music ones, so we are not bypassing nothing, just getting Spotify metadata, searching it on Youtube and downloading from there.

I wanted to do an Android app based on Spot-dl library, but actually, no Android wrapper exists, so maybe in a future, when i have more free time, I'll probably make a library wrapper for Android and also a GUI app like this. Working on it with Junkfood would be nice, but i don't want to disturb anyone.

Spotiflyer is actually kind of bad. It has a lot of problems downloading playlists, but it does it's work.

**Reply by RevealedSoulEven** [reply]
Yes, this point is correct coz I know spotiflyer gives error
